ocaml/otherlibs/stdlib_upstream_compatible/float_u.ml

```diff
@@ -19,9 +19,9 @@ open! Stdlib
 
 type t = float#
 
-external to_float : t -> (float[@local_opt]) = "%box_float"
+external to_float : t -> (float[@local_opt]) = "%box_float" [@@warning "-187"]
```

Comment:
Actually, I think we can (confusingly) compile the upstream-compatible otherlib with `stable`. The upstream-compatible otherlib will never actually get compiled with the upstream compiler. So it can use all manner of non-upstream-compatible features quite happily. Of course, its interface needs to be shared with a shim library, but that's a separate concern.

Reply:
Huh, you're right. I'll still leave it at `upstream_compatible` for now, but I'll think about it if we need to add more stuff.

Reply:
Actually, since we want to generate the shim interface by erasing the actual interface, probably best to leave it at `upstream_compatible`.

Reply:
Up to you as this evolves, but this suggests, to me, just having `[@@@warning "-187"]` right at the top of the .ml files (with a comment explaining why this is OK). No need to do that now, just planting a seed for later. :)